pytket/tests/compilation_test.py

```diff
@@ -154,67 +157,85 @@ def test_compilation() -> None:
     assert FullPeepholeOptimise().apply(c)
 
 
-def test_superpass() -> None:
+def test_PassSelector() -> None:
     fp = FullPeepholeOptimise()
     fp2 = FullPeepholeOptimise(allow_swaps=False)
 
-    sp = superpass([fp, fp2])
+    def circ_depth(circ: Circuit) -> int:
+        return circ.depth()
+
+    sp = PassSelector([fp, fp2], circ_depth)
 
     circ = Circuit(2).H(1).H(0).H(1).H(0).X(1).CX(1, 0).CX(0, 1).CX(1, 0)
 
     result = sp.apply(circ)
 
     assert sp.get_scores() == [1, 4]
 
-    assert result.depth() == min(sp.get_scores())
+    assert result.depth() == min([x for x in sp.get_scores() if x is not None])
```

Comment:
```suggestion
    assert result.depth() == min(x for x in sp.get_scores() if x is not None)
```
(and similarly below...)

Reply:
I have done the two below as well